src/app/models/Product.model.ts

```diff
@@ -0,0 +1,11 @@
+export interface Product {
+  uid: string; // This is an auto-generated field but we are including this for lisibility sake, will be set to the id of the document
+  images: Array<string>;
+  title: string;
+  price: number;
+  salesLevel: number; // % level of a sales on a product, the range should be from 0(full price) to 100(free)%
+  productType: string;
+  productDescription: string;
+  options: Map<string, string>;
```

Comment:
```suggestion
  options: {key: string, value: string};
```
This will be easier for the front